File: labs/1st course/2nd semestr/lab4/ts/rhombus.ts/string.ts
export class TextLine {
    protected _value: string;

    constructor();
    constructor(value: string);
    constructor(value?: string) {
        this._value = value ?? "";
    }

    get value(): string {
        return this._value;
    }

    length(): number {
        return this._value.length;
    }
}

export class DigitLine extends TextLine {
    constructor();
    constructor(value: string);
    constructor(value?: string) {
        super(value);
    }

    removeSymbol(symbol: string): void {
        this._value = this._value.replace(symbol, "");
    }

    get cleanedValue(): string {
        return this._value;
    }
}

File: labs/1st course/2nd semestr/lab4/cs/string.cs
using System;
using System.Text.RegularExpressions;

namespace LineLibrary
{
    public class TextLine
    {
        protected string _value;

        public TextLine()
        {
            _value = string.Empty;
        }

        public TextLine(string value)
        {
            _value = value;
        }

        public string Value
        {
            get { return _value; }
        }

        public int Length()
        {
            return _value.Length;
        }
    }

    public class DigitLine : TextLine
    {
        public DigitLine() : base() { }

        public DigitLine(string value) : base(value) { }

        public void RemoveSymbol(char symbol)
        {
            _value = Regex.Replace(_value, symbol.ToString(), "");
        }

        public string CleanedValue
        {
            get { return _value; }
        }
    }
}
